adlfs/multithread.py

```diff
@@ -135,7 +135,7 @@ def touch(self, src, dst):
             with open(dst, 'wb'):
                 pass
 
-        self.client.run(nthreads, monitor, before_scatter=touch)
+        self.client.run(nthreads, monitor, before_start=touch)
```

Comment:
Concatenates file to zero - results in one extra REST call per remote final file, correct? Checking for the existence of the file upfront would be one REST call or zero, if the file list was already cached.

btw: please check, is `adl.invalidate_cache(rfile)` being called when a file upload is finalised?